Sources/Core/AST/Stmt/ConditionalCompilationStmt.swift

```diff
@@ -94,10 +94,63 @@ public struct ConditionalCompilationStmt: Stmt {
 
   }
 
+  /// Abstracts the whole condition structure and its sub-conditions.
+  public indirect enum ConditionTree: Codable {
+
+    case operand(Condition)
+
+    case and(ConditionTree, ConditionTree)
+
+    case or(ConditionTree, ConditionTree)
+
+    public enum ConditionKind: String {
+      case holdOnly
+      case skipMain
+      case skipElse
+    }
+
+    /// Visit the SequenceCondition tail to calculate the right case for executing the right branch.
+    private func unroll(
+      for factors: ConditionalCompilationFactors, conditionCase: ConditionKind
+    ) -> Bool {
+      switch self {
+      case .and(let leftSubtree, let rightSubtree):
+        return leftSubtree.unroll(for: factors, conditionCase: conditionCase)
+          && rightSubtree.unroll(for: factors, conditionCase: conditionCase)
+
+      case .or(let leftSubtree, let rightSubtree):
+        return leftSubtree.unroll(for: factors, conditionCase: conditionCase)
+          || rightSubtree.unroll(for: factors, conditionCase: conditionCase)
+
+      case .operand(let cond):
+        switch conditionCase {
+        case .holdOnly: return cond.holds(for: factors)
+        case .skipMain: return cond.mayNotNeedParsing && !cond.holds(for: factors)
+        case .skipElse: return cond.mayNotNeedParsing && cond.holds(for: factors)
+        }
+      }
+    }
+
+    /// Return `true` iff the the full condition is satisfied.
+    public func mustSkipMainBranch(for factors: ConditionalCompilationFactors) -> Bool {
+      self.unroll(for: factors, conditionCase: ConditionKind.skipMain)
+    }
+
+    /// Return `true` iff the the full condition is unsatisfied.
+    public func mustSkipElseBranch(for factors: ConditionalCompilationFactors) -> Bool {
+      self.unroll(for: factors, conditionCase: ConditionKind.skipElse)
+    }
+
+    /// Return `true` iff the the full condition is satisfied.
```

Comment:
```suggestion
    /// Returns `true` iff the the condition is satisfied.
```
What is the difference between `mustSkipMainBranch(for:)` and `holds(for:)`?

Reply:
Originally I wanted to separated the intents to make more clear the proper action. On this of course I can also remove the 3 specific functions and directly use `unroll` and renaming the latter as `visitWithConditionKind`